### PR TITLE
Add support of Rails 5.1.0

### PR DESCRIPTION
**Pull request by n-rodriguez**
Rails is currently in beta1 and everything is fine.

### CODE REVIEW

**Comment by n-rodriguez** [comment]
Hi! Tests are ok. Maybe we could remove support of Rails 4.0.x and 4.1.x?
What do you think?

**Comment by riqwan** [comment]
@n-rodriguez - Is this ready to be merged?

**Comment by n-rodriguez** [comment]
Yes!

**Comment by jonkessler** [comment]
I don't see much of a need to remove the 4.0 and 4.1 code yet. I'll take a look at the 5.1 code tomorrow hopefully.

File: lib/store_base_sti_class_for_5_1.rb [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+              ([source_type.constantize] + source_type.constantize.descendants).map(&:to_s)
+            end
+          # END PATCH
+
+          lambda { |object| where(type => adjusted_source_type) }
+        end
+      end
+
+      class ThroughReflection
+        def scope_chain
```
Comment (jonkessler) [inline lib/store_base_sti_class_for_5_1.rb]: It looks like this is gone in 5.1 as `scope_chain` is now deprecated.

Comment (n-rodriguez) [inline lib/store_base_sti_class_for_5_1.rb]: Removed

**Comment by n-rodriguez** [comment]
> Left one comment. Is there a way to share the code that's the same between the 5.0 and 5.1 patches? There's a lot of overlap.

I do agree but I would prefer to keep it simple. It's easier to resonate on a single file and to compare 2 files. Sometimes a good duplication is better than a bad abstraction and I think this is the case.

**Pull request merged**